File: dotnet/HotelBooking/HotelBooking/Models/Room.cs
namespace HotelBooking.Models;

public class Room
{
    private static readonly Dictionary<RoomType, int> Prices = new Dictionary<RoomType, int>
    {
        { RoomType.Suite, 10000 },
        { RoomType.Business, 5000 },
        { RoomType.Standard, 2500 },
        { RoomType.Basic, 1000 }
    };

    public Room(RoomType type)
    {
        Type = type;
    }

    public Room() {}

    public int Id { get; set; }
    public RoomType Type { get; set; }
    public List<Booking> Bookings { get; set; } = new();

    public int RoomPrice => Prices[Type];

    public int TotalEarnings ()  => Bookings.Sum(booking => booking.Bill);
}

public enum RoomType
{
    Suite, Business, Standard, Basic
}

File: java/hotelbooking/src/main/java/no/egde/hotelbooking/models/Room.java
package no.egde.hotelbooking.models;

import javax.persistence.*;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

@Entity
public class Room {
    private static final Map<RoomType, Integer> prices = new HashMap<>(Map.ofEntries(
        Map.entry(RoomType.SUITE, 10000),
        Map.entry(RoomType.BUSINESS, 5000),
        Map.entry(RoomType.STANDARD, 2500),
        Map.entry(RoomType.BASIC, 1000)
    ));

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Enumerated(EnumType.STRING)
    private RoomType roomType;

    public Room(RoomType type) {
        this.roomType = type;
    }

    public Room() {

    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public RoomType getRoomType() {
        return roomType;
    }

    public void setRoomType(RoomType roomType) {
        this.roomType = roomType;
    }

    public Integer getRoomPrice() {
        return prices.get(roomType);
    }
}
